// symphony-integration-commons/dist/components/SubmitInstance/SubmitInstanceContainer.js
'use strict';

Object.defineProperty(exports, "__esModule", {
  value: true
});

var _reactRedux = require('react-redux');

var _actions = require('../../actions');

var _SubmitInstance = require('./SubmitInstance');

var _SubmitInstance2 = _interopRequireDefault(_SubmitInstance);

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

var store = {};

var mapStateToProps = function mapStateToProps(state) {
  store = Object.assign({}, state.instance);
  return {
    saved: state.instance.saved,
    name: state.instance.name,
    streamType: state.instance.streamType,
    postingRooms: state.instance.streams
  };
};

var mapDispatchToProps = function mapDispatchToProps(dispatch) {
  return {
    saveInstance: function saveInstance() {
      dispatch((0, _actions.saveInstance)(store));
    },
    editInstance: function editInstance() {
      dispatch((0, _actions.editInstance)(store));
    },
    removeInstance: function removeInstance() {
      dispatch((0, _actions.removeInstance)(store));
    },
    resetMessage: function resetMessage() {
      dispatch((0, _actions.resetMessage)());
    },
    showRequireName: function showRequireName() {
      dispatch((0, _actions.showRequireName)());
    },
    showRequireRooms: function showRequireRooms() {
      dispatch((0, _actions.showRequireRooms)());
    },
    hideRequireName: function hideRequireName() {
      dispatch((0, _actions.hideRequireName)());
    },
    hideRequireRooms: function hideRequireRooms() {
      dispatch((0, _actions.hideRequireRooms)());
    }
  };
};

var SubmitInstanceContainer = (0, _reactRedux.connect)(mapStateToProps, mapDispatchToProps)(_SubmitInstance2.default);

exports.default = SubmitInstanceContainer;
