// symphony-integration-commons/dist/samples/posting-location-sample/components/CustomCreate.js
'use strict';

Object.defineProperty(exports, "__esModule", {
  value: true
});

var _react = require('react');

var _react2 = _interopRequireDefault(_react);

var _reactRouter = require('react-router');

var _symphonyIntegrationCommons = require('symphony-integration-commons');

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

var styles = {
  header: {
    paddingLeft: 15,
    maxWidth: 840
  },
  content: {
    marginBottom: 30
  },
  sub: {
    paddingTop: 20,
    paddingLeft: 15,
    maxWidth: 840
  }
};

var submit = function submit() {
  _reactRouter.hashHistory.push('/choose-stream-type');
};

var cancel = function cancel() {
  _reactRouter.hashHistory.push('/');
};

var CustomCreate = function CustomCreate() {
  return _react2.default.createElement(
    'div',
    null,
    _react2.default.createElement(
      'header',
      { style: styles.header },
      _react2.default.createElement(
        'h1',
        null,
        'My Custom Integration'
      ),
      _react2.default.createElement(
        'div',
        { style: styles.content },
        _react2.default.createElement(
          'p',
          null,
          'This is my custom configurator App. Please, create an instance.'
        )
      )
    ),
    _react2.default.createElement(_symphonyIntegrationCommons.InputDescription, null),
    _react2.default.createElement(
      'div',
      { style: styles.sub },
      _react2.default.createElement(
        'button',
        { className: 'button cancel-link', onClick: function onClick() {
            cancel();
          } },
        'Cancel'
      ),
      _react2.default.createElement(
        'button',
        { className: 'button', onClick: function onClick() {
            submit();
          } },
        'Next'
      )
    )
  );
};

exports.default = CustomCreate;
