// symphony-integration-commons/dist/samples/posting-location-sample/components/CustomInstanceCreated.js
'use strict';

Object.defineProperty(exports, "__esModule", {
  value: true
});

var _react = require('react');

var _react2 = _interopRequireDefault(_react);

var _reactRouter = require('react-router');

var _symphonyIntegrationCommons = require('symphony-integration-commons');

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

var styles = {
  header: {
    paddingLeft: 15,
    maxWidth: 840
  },
  sub: {
    paddingTop: 20,
    paddingLeft: 15,
    maxWidth: 840
  }
};

var submit = function submit() {
  _reactRouter.hashHistory.push('/');
};

var CustomInstanceCreated = function CustomInstanceCreated() {
  return _react2.default.createElement(
    'div',
    null,
    _react2.default.createElement(
      'header',
      { style: styles.header },
      _react2.default.createElement(
        'h1',
        null,
        'My Custom Integration'
      ),
      _react2.default.createElement(
        'h2',
        null,
        'Webhook URL'
      )
    ),
    _react2.default.createElement(_symphonyIntegrationCommons.WebHookURL, null),
    _react2.default.createElement(
      'div',
      { style: styles.sub },
      _react2.default.createElement(
        'button',
        { className: 'button', onClick: function onClick() {
            submit();
          } },
        'Done'
      )
    )
  );
};

exports.default = CustomInstanceCreated;
