// symphony-integration-commons/dist/samples/posting-location-sample/components/InstanceList.js
'use strict';

Object.defineProperty(exports, "__esModule", {
  value: true
});

var _react = require('react');

var _react2 = _interopRequireDefault(_react);

var _reactRouter = require('react-router');

var _symphonyIntegrationCommons = require('symphony-integration-commons');

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

var styles = {
  header: {
    paddingLeft: 15,
    maxWidth: 840
  },
  content: {
    marginBottom: 30
  },
  sub: {
    paddingTop: 20,
    paddingLeft: 15,
    maxWidth: 840
  }
};

var create = function create() {
  _reactRouter.hashHistory.push('/custom-create');
};

var InstanceList = function InstanceList() {
  return _react2.default.createElement(
    'div',
    null,
    _react2.default.createElement(
      'header',
      { style: styles.header },
      _react2.default.createElement(
        'h1',
        null,
        'My Custom Integration'
      ),
      _react2.default.createElement(
        'div',
        { style: styles.content },
        _react2.default.createElement(
          'p',
          null,
          'Lorem ipsum dolor sit amet, consectetur adipiscing elit. Nulla tincidunt a enim quis sagittis. Sed auctor nulla efficitur sollicitudin ullamcorper. Suspendisse faucibus velit et orci aliquam, quis pulvinar dui lobortis. Vivamus mattis tortor felis, vitae tristique libero dapibus eget. Vivamus gravida elit non varius facilisis. Mauris ac metus at erat facilisis laoreet. Morbi nec dignissim enim. Pellentesque vitae dui scelerisque, cursus magna interdum, lobortis lectus. Etiam convallis magna nec purus mattis volutpat. Cras tincidunt lorem nisl, sit amet mattis ex molestie placerat.'
        )
      )
    ),
    _react2.default.createElement(_symphonyIntegrationCommons.TableInstance, null),
    _react2.default.createElement(
      'div',
      { style: styles.sub },
      _react2.default.createElement(
        'button',
        { className: 'button', onClick: function onClick() {
            create();
          } },
        'Create Instance'
      )
    )
  );
};

exports.default = InstanceList;
