// symphony-integration-commons/dist/components/InputDescriptionInfo/InputDescriptionInfo.js
'use strict';

Object.defineProperty(exports, "__esModule", {
  value: true
});

var _react = require('react');

var _react2 = _interopRequireDefault(_react);

var _reactRedux = require('react-redux');

require('../InputDescription/styles/styles.less');

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

var InputDescriptionInfo = function InputDescriptionInfo(_ref) {
  var name = _ref.name;
  return _react2.default.createElement(
    'div',
    { className: 'wrapper input-description' },
    _react2.default.createElement(
      'header',
      null,
      _react2.default.createElement(
        'h2',
        null,
        _react2.default.createElement(
          'label',
          { htmlFor: 'input-description' },
          'Description'
        )
      )
    ),
    _react2.default.createElement('input', {
      type: 'text',
      className: 'text-input',
      id: 'input-description',
      defaultValue: name,
      disabled: 'disabled'
    })
  );
}; /* eslint-disable no-unused-vars */


InputDescriptionInfo.propTypes = {
  name: _react.PropTypes.string.isRequired
};

var mapStateToProps = function mapStateToProps(state) {
  return {
    name: state.instance.name
  };
};

exports.default = (0, _reactRedux.connect)(mapStateToProps)(InputDescriptionInfo);
