// symphony-integration-commons/dist/components/PostingLocation/SuggestionsRooms/SuggestionsRoomsContainer.js
'use strict';

Object.defineProperty(exports, "__esModule", {
  value: true
});

var _reactRedux = require('react-redux');

var _actions = require('../../../actions/');

var _SuggestionsRooms = require('./SuggestionsRooms');

var _SuggestionsRooms2 = _interopRequireDefault(_SuggestionsRooms);

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

var mapStateToProps = function mapStateToProps(state) {
  return {
    userRooms: state.userRooms,
    filters: state.instance.postingLocationRooms
  };
};

var mapDispatchToProps = function mapDispatchToProps(dispatch) {
  return {
    addStreamToInstance: function addStreamToInstance(room) {
      dispatch((0, _actions.addStreamToInstance)(room));
    },
    removeStreamFromInstance: function removeStreamFromInstance(room) {
      dispatch((0, _actions.removeStreamFromInstance)(room));
    },
    resetPostingLocation: function resetPostingLocation() {
      dispatch((0, _actions.resetPostingLocationRooms)());
    }
  };
};

var SuggestionsRoomsContainer = (0, _reactRedux.connect)(mapStateToProps, mapDispatchToProps)(_SuggestionsRooms2.default);

exports.default = SuggestionsRoomsContainer;
