// symphony-integration-commons/dist/components/TableInstance/TableInstanceContainer.js
'use strict';

Object.defineProperty(exports, "__esModule", {
  value: true
});

var _reactRedux = require('react-redux');

var _actions = require('../../actions');

var _TableInstance = require('./TableInstance');

var _TableInstance2 = _interopRequireDefault(_TableInstance);

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

var mapStateToProps = function mapStateToProps(state) {
  return {
    userId: state.userId,
    appName: state.appParams.appName,
    rooms: state.userRooms,
    instanceList: state.instanceList.instances,
    firstScreen: state.instanceList.firstScreen,
    loading: state.instanceList.loading,
    typeMessage: state.message.type,
    baseWebHookURL: state.instance.baseWebHookURL
  };
};

var mapDispatchToProps = function mapDispatchToProps(dispatch) {
  return {
    getInstanceList: function getInstanceList() {
      dispatch((0, _actions.getInstanceList)());
    },
    showEditInstanceView: function showEditInstanceView(_instance) {
      dispatch((0, _actions.getInstanceInfo)(_instance));
    },
    showNoInstancesFound: function showNoInstancesFound() {
      dispatch((0, _actions.showNoInstancesFound)());
    },
    resetMessage: function resetMessage() {
      dispatch((0, _actions.resetMessage)());
    }
  };
};

var TableInstanceContainer = (0, _reactRedux.connect)(mapStateToProps, mapDispatchToProps)(_TableInstance2.default);

exports.default = TableInstanceContainer;
