// symphony-integration-commons/dist/reducers/index.js
'use strict';

Object.defineProperty(exports, "__esModule", {
  value: true
});

var _redux = require('redux');

var _userReducer = require('./userReducer');

var _userReducer2 = _interopRequireDefault(_userReducer);

var _instancesListReducer = require('./instancesListReducer');

var _instancesListReducer2 = _interopRequireDefault(_instancesListReducer);

var _roomsReducer = require('./roomsReducer');

var _roomsReducer2 = _interopRequireDefault(_roomsReducer);

var _paramsReducer = require('./paramsReducer');

var _paramsReducer2 = _interopRequireDefault(_paramsReducer);

var _instructionsReducer = require('./instructionsReducer');

var _instructionsReducer2 = _interopRequireDefault(_instructionsReducer);

var _instanceReducer = require('./instanceReducer');

var _instanceReducer2 = _interopRequireDefault(_instanceReducer);

var _messagesReducer = require('./messagesReducer');

var _messagesReducer2 = _interopRequireDefault(_messagesReducer);

var _errorReducer = require('./errorReducer');

var _errorReducer2 = _interopRequireDefault(_errorReducer);

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

var rootReducer = (0, _redux.combineReducers)({
  appParams: _paramsReducer2.default,
  appInstructions: _instructionsReducer2.default,
  userId: _userReducer2.default,
  userRooms: _roomsReducer2.default,
  instanceList: _instancesListReducer2.default,
  instance: _instanceReducer2.default,
  message: _messagesReducer2.default,
  error: _errorReducer2.default
});

exports.default = rootReducer;
