// symphony-integration-commons/dist/components/PostingLocation/RoomBox/RoomBox.js
'use strict';

Object.defineProperty(exports, "__esModule", {
  value: true
});

var _react = require('react');

var _react2 = _interopRequireDefault(_react);

require('./styles/styles.less');

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

var RoomBox = function RoomBox(props) {
  return _react2.default.createElement(
    'div',
    { className: 'room-box' },
    _react2.default.createElement(
      'div',
      null,
      _react2.default.createElement(
        'span',
        null,
        props.name
      ),
      !props.public && _react2.default.createElement(
        'span',
        null,
        _react2.default.createElement('i', { className: 'fa fa-lock' })
      ),
      _react2.default.createElement(
        'div',
        { className: 'room-info' },
        _react2.default.createElement(
          'span',
          null,
          props.memberCount + ' Member' + (props.memberCount > 1 ? 's' : '') + ',\n              created by ' + props.creatorPrettyName
        )
      )
    ),
    _react2.default.createElement(
      'button',
      {
        onClick: function onClick() {
          return props.removeFilter(props.threadId);
        }
      },
      _react2.default.createElement('i', { className: 'fa fa-times' })
    )
  );
};

RoomBox.propTypes = {
  name: _react.PropTypes.string.isRequired,
  threadId: _react.PropTypes.string.isRequired,
  public: _react.PropTypes.bool.isRequired,
  memberCount: _react.PropTypes.number.isRequired,
  creatorPrettyName: _react.PropTypes.string.isRequired,
  removeFilter: _react.PropTypes.func.isRequired
};

exports.default = RoomBox;
